warehouse/query-core/src/test/java/datawave/query/iterator/SortedListKeyValueIterator.java

```diff
@@ -38,6 +39,14 @@ public SortedListKeyValueIterator(List<Map.Entry<Key,Value>> sourceList) {
         currentIndex = 0;
     }
     
+    public SortedListKeyValueIterator(SortedMap<Key,Value> map) {
+        this.sourceList = new ArrayList<>(map.entrySet().size());
+        currentIndex = 0;
+        for (Map.Entry<Key,Value> entry : map.entrySet()) {
+            sourceList.add(entry);
+        }
```

Comment:
Could this be a sourceList.addAll(map.entrySet())?
Does the currentIndex indicate another possible constructor where we preserve the current index?